.github/workflows/preview.yml

```diff
@@ -0,0 +1,48 @@
+name: Surge.sh Preview
+
+on:
+  workflow_run:
+    workflows: ["Build Adoc"]
+    types:
+      - completed
+
+jobs:
+  preview:
+    runs-on: ubuntu-latest
+    if: github.event.workflow_run.event == 'pull_request' && github.event.workflow_run.conclusion == 'success'
+    steps:
+      - name: Download PR Artifact
+        uses: dawidd6/action-download-artifact@v2
+        with:
+          workflow: ${{ github.event.workflow_run.workflow_id }}
+          workflow_conclusion: success
+          name: site
+      - name: Store PR id as variable
+        id: pr
+        run: |
+          echo "id=$(<pr-id.txt)" >> $GITHUB_OUTPUT 
```

Comment:
We need this file generated either in build or Build Adoc. 

@gastaldi I guess we can save this file in Build Adoc with:

```
    - name: Store PR id
      run: |
        echo ${{ github.event.number }} > pr-id.txt
```

This workflow is slightly different from the one you posted here: 

https://github.com/quarkiverse/quarkus-prettytime/blob/main/.github/workflows/build.yml#L43C8-L52

@luanbrdev Can you try running locally your PR and see if we get the `./docs/target/generated-docs` dir?

Then, I think we can use the same approach and have the website built in the maven build process instead. Here's the reference: https://github.com/quarkiverse/quarkus-prettytime/tree/main/.github/workflows

Reply:
Hi, what's up?! I will use the reference. But, I have a question, in the build.yml file in our project we have a path-ignore for adoc files (if I'm understand correctly). When I created build-adoc.yml I was think about only run the preview in cases of adoc files update. This make sense? 
About run the actions locally, I dunno how to do this, but I will search about. :stuck_out_tongue_winking_eye:

Reply:
> When I created build-adoc.yml I was think about only run the preview in cases of adoc files update. This make sense?

Yes, indeed! You're right, we can keep both workflows separate. 

> About run the actions locally, I dunno how to do this, but I will search about

You can mimic what the workflow is doing. 👍